packages/pigeon/lib/gobject_generator.dart

```diff
@@ -0,0 +1,2071 @@
+    return 'void';
+  } else if (type.baseName == 'bool') {
+    return type.isNullable && !primitive ? 'gboolean*' : 'gboolean';
+  } else if (type.baseName == 'int') {
+    return type.isNullable && !primitive ? 'int64_t*' : 'int64_t';
+  } else if (type.baseName == 'double') {
+    return type.isNullable && !primitive ? 'double*' : 'double';
+  } else if (type.baseName == 'String') {
+    return isOutput ? 'gchar*' : 'const gchar*';
+  } else if (type.baseName == 'Uint8List') {
+    return isOutput ? 'uint8_t*' : 'const uint8_t*';
+  } else if (type.baseName == 'Int32List') {
+    return isOutput ? 'int32_t*' : 'const int32_t*';
+  } else if (type.baseName == 'Int64List') {
+    return isOutput ? 'int64_t*' : 'const int64_t*';
+  } else if (type.baseName == 'Float32List') {
+    return isOutput ? 'float*' : 'const float*';
+  } else if (type.baseName == 'Float64List') {
+    return isOutput ? 'double*' : 'const double*';
+  } else {
+    throw Exception('Unknown type ${type.baseName}');
+  }
+}
+
+// Returns true if [type] is a *List typed numeric list type.
+bool _isNumericListType(TypeDeclaration type) {
+  return type.baseName == 'Uint8List' ||
+      type.baseName == 'Int32List' ||
+      type.baseName == 'Int64List' ||
+      type.baseName == 'Float32List' ||
+      type.baseName == 'Float64List';
+}
+
+// Returns true if [type] is a nullable type with a primitive native data type.
+bool _isNullablePrimitiveType(TypeDeclaration type) {
+  if (!type.isNullable) {
+    return false;
+  }
+
+  return type.isEnum ||
+      type.baseName == 'bool' ||
+      type.baseName == 'int' ||
+      type.baseName == 'double';
+}
+
+// Whether [type] is a type that needs to stay an FlValue* since it can't be
+// expressed as a more concrete type.
+bool _isFlValueWrappedType(TypeDeclaration type) {
+  return type.baseName == 'List' ||
+      type.baseName == 'Map' ||
+      type.baseName == 'Object';
+}
+
+// Returns code to clear a value stored in [variableName], or null if no function required.
+String? _getClearFunction(TypeDeclaration type, String variableName) {
+  if (type.isClass) {
+    return 'g_clear_object(&$variableName)';
+  } else if (_isFlValueWrappedType(type)) {
+    return 'g_clear_pointer(&$variableName, fl_value_unref)';
+  } else if (type.baseName == 'String') {
+    return 'g_clear_pointer(&$variableName, g_free)';
+  } else if (_isNullablePrimitiveType(type)) {
+    return 'g_clear_pointer(&$variableName, g_free)';
+  } else {
+    return null;
+  }
+}
+
+// Returns code for the default value for [type].
+String _getDefaultValue(String module, TypeDeclaration type,
+    {bool primitive = false}) {
+  if (type.isClass || (type.isNullable && !primitive)) {
+    return 'nullptr';
+  } else if (type.isEnum) {
+    final String enumName = _getClassName(module, type.baseName);
+    return 'static_cast<$enumName>(0)';
+  } else if (_isFlValueWrappedType(type)) {
+    return 'nullptr';
+  } else if (type.baseName == 'void') {
```

Comment:
Optional nit here and in the methods below: it might be clearer to do all the special cases, then instead of all the baseName if/else's do:

```dart
else {
  switch(type.baseName) {
    case 'foo': ...
    case 'bar': ...
    ...
  }
}
```